apps/api/v2/src/ee/event-types/event-types_2024_06_14/services/event-types.service.ts

```diff
@@ -0,0 +1,256 @@
+    if (!user) {
+      return [];
+    }
+    return this.getUserEventTypes(user.id);
+  }
+
+  async getUserToCreateEvent(user: UserWithProfile) {
+    const organizationId = user.movedToProfile?.organizationId || user.organizationId;
+    const isOrgAdmin = organizationId
+      ? await this.membershipsRepository.isUserOrganizationAdmin(user.id, organizationId)
+      : false;
+    const profileId = user.movedToProfile?.id || null;
+    return {
+      id: user.id,
+      organizationId: user.organizationId,
+      organization: { isOrgAdmin },
+      profile: { id: profileId },
+      metadata: user.metadata,
+    };
+  }
+
+  async getUserEventType(userId: number, eventTypeId: number) {
+    const eventType = await this.eventTypesRepository.getUserEventType(userId, eventTypeId);
+
+    if (!eventType) {
+      return null;
+    }
+
+    this.checkUserOwnsEventType(userId, eventType);
+    return this.outputEventTypesService.getResponseEventType(userId, eventType);
+  }
+
+  async getUserEventTypes(userId: number) {
+    const eventTypes = await this.eventTypesRepository.getUserEventTypes(userId);
+
+    const eventTypePromises = eventTypes.map(async (eventType) => {
+      return await this.outputEventTypesService.getResponseEventType(userId, eventType);
+    });
+
+    return await Promise.all(eventTypePromises);
+  }
+
+  async getUserEventTypeForAtom(user: UserWithProfile, eventTypeId: number) {
+    const organizationId = user.movedToProfile?.organizationId || user.organizationId;
+
+    const isUserOrganizationAdmin = organizationId
+      ? await this.membershipsRepository.isUserOrganizationAdmin(user.id, organizationId)
+      : false;
+
+    const eventType = await this.eventTypesRepository.getUserEventTypeForAtom(
+      user,
+      isUserOrganizationAdmin,
+      eventTypeId
+    );
+
+    if (!eventType) {
+      return null;
+    }
+
+    this.checkUserOwnsEventType(user.id, eventType.eventType);
+    return eventType;
+  }
+
+  async getEventTypesPublicByUsername(username: string): Promise<EventTypesPublic> {
+    const user = await this.usersRepository.findByUsername(username);
+    if (!user) {
+      throw new NotFoundException(`User with username "${username}" not found`);
+    }
+
+    return await getEventTypesPublic(user.id);
+  }
+
+  async getEventTypes(queryParams: GetEventTypesQuery_2024_06_14) {
+    const { username, eventSlug, usernames } = queryParams;
+    let eventTypes: EventTypeOutput_2024_06_14[] = [];
+
+    if (username && eventSlug) {
+      const eventType = await this.getEventTypeByUsernameAndSlug(username, eventSlug);
+      if (eventType) {
+        eventTypes.push(eventType);
+      }
+    } else if (username && !eventSlug) {
+      eventTypes = await this.getEventTypesByUsername(username);
+    } else if (usernames) {
+      const dynamicEventType = await this.getDynamicEventType(usernames);
+      eventTypes.push(dynamicEventType);
+    }
+
+    return eventTypes;
+  }
```

Comment:
I think we can make this more readable by using early returns

Reply:
Done in [refactor: simplify getEventTypes](https://github.com/calcom/cal.com/pull/15457/commits/2fbd738d43ddf63cb800ae9ab586aeeaf5ed04b6)

Reply:
very nicely done